### ISSUE TITLE
Paper formulas return ksi = 0

### ISSUE THREAD

**Issue by Vayel** [XAI-ANITI/ethik#125]
```python
import ethik
import lightgbm as lgb
import numpy as np
import pandas as pd
import plotly.graph_objs as go
from scipy import optimize
from sklearn import metrics, model_selection

X, y = ethik.datasets.load_adult()
X_train, X_test, y_train, y_test = model_selection.train_test_split(X, y, shuffle=True, random_state=42)

model = lgb.LGBMClassifier(random_state=42).fit(X_train, y_train)
y_pred = model.predict_proba(X_test)[:, 1]
y_pred = pd.Series(y_pred, name=">$50k")

def safe_scale(x):
    if isinstance(x, pd.DataFrame):
        return (x - x.mean()) / x.std().replace(0, 1)
    return (x - x.mean(axis=0)) / (x.std(axis=0) or 1)

class ConvergenceSuccess(Exception):
    def __init__(self, ksi):
        self.ksi = ksi

class F:
    def __init__(self, x, target_mean, tol):
        self.x = x
        self.target_mean = target_mean
        self.tol = tol

    def __call__(self, ksi):
        n = len(self.x)
        if len(ksi) == 1:
            print("dim 1")
            t = self.target_mean[0]
            x = self.x[:, 0]
            e = np.exp(ksi[0] * x)
            s = np.sum(e)
            print("e", e, "s", s)
            # When ksi == 0 (the starting point), e = 1, s = n
            # and log(s/n) = log(1) = 0, so the loss is 0
            loss = np.log(1 / n * s) - ksi[0] * t
            g = np.sum(x * e) / s - t
        else:
            t = self.target_mean
            x = self.x
            e = np.exp(np.dot(x, ksi))
            s = np.sum(e)
            loss = np.log(1 / n * s) - np.dot(ksi, t)
            g = np.sum(x * e) / s - t

        print("loss", loss, "g", g)
        if loss < self.tol:
            raise ConvergenceSuccess(ksi=ksi)

        return loss, g


def compute_ksi(group_id, x, target_mean, max_iterations, tol):
    mean, std = x.mean().values, x.std().values
    features = x.columns
    x = safe_scale(x).values
    ksis = np.zeros(len(target_mean))

    if np.isclose(target_mean, mean, atol=tol).all():
        return {(group_id, feature): (0.0, True) for feature in features}

    converged = False
    try:
        res = optimize.minimize(
            fun=F(x=x, target_mean=(target_mean - mean) / std, tol=tol),
            x0=ksis,  # Initial ksi value
            jac=True,
            method="BFGS",
        )
    except ConvergenceSuccess as cs:
        ksis = cs.ksi
        converged = True

    return {
        (group_id, feature): (ksi, converged) for feature, ksi in zip(features, ksis)
    }
```

```python
compute_ksi(
    group_id=0,
    x=X_test[["age"]],
    target_mean=[30],
    max_iterations=15,
    tol=1e-4,
)
```

```
dim 1
e [1. 1. 1. ... 1. 1. 1.] s 8141.0
loss 0.0 g 0.6252068973348669

{(0, 'age'): (0.0, True)}
```

**Comment by Vayel** [comment]
Here is the loss function given by the paper:

![Screenshot from 2019-12-11 11-06-04](https://user-images.githubusercontent.com/6124369/70611944-4dde0980-1c06-11ea-89f0-91537626e6e7.png)

and the gradient:

![Screenshot from 2019-12-11 11-24-16](https://user-images.githubusercontent.com/6124369/70613386-d198f580-1c08-11ea-8bb2-284b24bbeeef.png)

Here is the loss function we currently have:

![image](https://user-images.githubusercontent.com/6124369/70613650-4c621080-1c09-11ea-8594-e2acb6dd8dea.png)

and the gradient:

![image](https://user-images.githubusercontent.com/6124369/70613763-8206f980-1c09-11ea-8524-67a968f1ea31.png)

**Comment by Vayel** [comment]
@MaxHalford @lrisser it's weird, our loss function is more or less the gradient in the paper.

**Comment by Vayel** [comment]
One difference with our current loss: the loss in the paper is not positive, which makes it difficult for us to determine a stop criterion.

**Comment by Vayel** [comment]
Here is an example with an overflow (see first message for function definitions):

```python
compute_ksi(
    x=X_test[["age", "education-num"]],
    target_mean=[19, 17],
    max_iterations=5,
    tol=1e-4,
)
```
